Sources/StreamChat/ChatClient.swift

```diff
@@ -441,15 +449,24 @@ public class ChatClient {
         }
     }
 
+    /// Cancels any hanging requests and switches to a new user
+    /// - Parameter token: The token for the new user
     func switchToNewUser(with token: Token) {
         completeTokenWaiters(token: nil)
         updateUser(with: token, completeTokenWaiters: false, isFirstConnection: false)
     }
 
+    /// Updates the WebSocket endpoint to use the passed token and user information for the connection
     func updateWebSocketEndpoint(with token: Token, userInfo: UserInfo?) {
         webSocketClient?.connectEndpoint = .webSocketConnect(userInfo: userInfo ?? .init(id: token.userId))
     }
 
+    /// Clears state related to the current user to leave the client ready for another user
+    /// Will clear:
+    ///     - Background workers
+    ///     - References to active controllers
+    ///     - Database
+    /// - Parameter completion: A block to be executed when the process is completed. Contains an error if something went wrong
     func clearPreviousUserData(completion: @escaping (Error?) -> Void) {
```

Comment:
Isn't `clearCurrentUserData` actually more suited ? 🤔 It will be the "previous" user, but when we call this, it is still the current user I guess 😄 And I can see this being used for the `last_sync_at` issue that we have at the moment. And in that case, this name will not make sense